cairo_zero/kakarot/library.cairo

```diff
@@ -160,21 +160,38 @@ namespace Kakarot {
     }
 
     // @notice Set the block base fee.
+    // @dev There can only be one base fee for a given block. Thus, we use an index to manage two different base fees:
+    // - The base fee to use for the current block (index: 'current_block')
+    // - The base fee to use for the next block (index: 'next_block')
     // @param base_fee The new base fee.
     func set_base_fee{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
         base_fee: felt
     ) {
-        Kakarot_base_fee.write(base_fee);
+        alloc_locals;
+        let (block_number) = get_block_number();
+        let (res_next_block) = Kakarot_base_fee.read('next_block');
+        let next_base_fee = res_next_block[0];
+        let starting_block = res_next_block[1];
+        Kakarot_base_fee.write('next_block', (base_fee, block_number + 1));
+
+        let is_next_fee_ready = is_le(starting_block, block_number);
+        if (is_next_fee_ready == FALSE) {
+            return ();
+        }
+
+        Kakarot_base_fee.write('current_block', (base_fee, starting_block));
```

Comment:
```suggestion
```

you cannot set base fee directly, otherwise the whole PR is somehow pointless as you can submit a set_base_fee with prvious block numbers and have it performed directly.

Reply:
> you can submit a set_base_fee with prvious block numbers

wdym? `set_base_fee` is always going to use `block_number + 1`. What this does is simply write the value of `next_block` in `current_block` if the value in `next_block` is already "active"; but no one has called `get_base_fee` in between.

Reply:
so maybe you wanted to do 

```
Kakarot_base_fee.write('current_block', (next_base_fee, starting_block));
```

but this just shouldn't be here. Why would you add in a `set_base_fee` an logic regarding `get_base_fee`?

Reply:
> Why would you add in a set_base_fee a logic regarding get_base_fee?

Because if you do

block=n-1: `current_base_fee=z`
block=n: `set_base_fee(x)`; `current_base_fee=z`
block=n+1: set_base_fee(y); `current_base_fee=z` (because no one has called `get_base_fee` yet

you would expect that at block=`n+1`, `current_base_fee=x`, but because you overwrote the `next_base_fee`, the value was lost and `current_base_fee` was not updated